dice-mfg/src/lib.rs

```diff
@@ -62,162 +62,140 @@ impl fmt::Display for Error {
     }
 }
 
-// https://github.com/oxidecomputer/dice-util/issues/16
-#[allow(clippy::too_many_arguments)]
-pub fn do_manufacture(
-    port: &mut Box<dyn SerialPort>,
-    openssl_cnf: PathBuf,
-    ca_section: Option<String>,
-    v3_section: Option<String>,
-    engine_section: Option<String>,
-    ping_retry: u8,
-    platform_id: PlatformId,
-    intermediate_cert: PathBuf,
-    no_yubi: bool,
-) -> Result<()> {
-    do_liveness(port, ping_retry)?;
-    do_set_platform_id(port, platform_id)?;
+/// The MfgDriver is used to send commands to the RoT as part of programming
+/// identity credentials. The structure holds SerialPort instance. The
+/// associated functions map 1:1 to members of the MfgMessage enum from
+/// dice-mfg-msgs. The `liveness` function is a minor exception to this rule.
+pub struct MfgDriver {
+    port: Box<dyn SerialPort>,
+}
 
-    let temp_dir = tempfile::tempdir()?;
-    let csr = Some(temp_dir.path().join("csr.pem"));
-    do_get_csr(port, &csr)?;
+impl MfgDriver {
+    pub fn new(port: Box<dyn SerialPort>) -> Self {
+        MfgDriver { port }
+    }
 
-    let cert = temp_dir.into_path().join("cert.pem");
-    do_sign_cert(
-        &cert,
-        &openssl_cnf,
-        ca_section,
-        v3_section,
-        engine_section,
-        &csr.unwrap(),
-        no_yubi,
-    )?;
-    do_set_device_id(port, &cert)?;
-    do_set_intermediate(port, &intermediate_cert)?;
-    do_break(port)
-}
-pub fn do_break(port: &mut Box<dyn SerialPort>) -> Result<()> {
-    print!("sending Break ... ");
-    match send_break(port) {
-        Ok(_) => {
-            println!("success");
-            Ok(())
-        }
-        Err(e) => {
-            println!("failed");
-            Err(e)
+    /// Ping the RoT at most `max_fail` times. If the RoT does not reply with
+    /// an Ack to one of these Pings this function returns an error.
+    pub fn liveness(&mut self, mut max_fail: u8) -> Result<()> {
+        print!("checking RoT for liveness ... ");
+        io::stdout().flush()?;
+
+        loop {
+            send_msg(&mut self.port, &MfgMessage::Ping)?;
+            match recv_ack(&mut self.port) {
+                Err(e) => {
+                    if max_fail > 0 {
+                        max_fail -= 1;
+                    } else {
+                        return Err(e);
+                    }
+                }
+                _ => {
```

Comment:
I'd have found Ok(()) to be clearer here than there is exactly one expected response.  Instead, I went looking to see what recv_ack returns.